gtest/src/manager.rs

```diff
@@ -419,6 +424,33 @@ impl ExtManager {
 
         core_processor::handle_journal(journal, self);
     }
+
+    pub(crate) fn get_mailbox(&self, program_id: &ProgramId) -> Vec<Message> {
+        match self.actor_to_mailbox.get(program_id) {
+            None => {
+                vec![]
+            }
+            Some(messages) => messages.clone(),
+        }
+    }
+
+    pub(crate) fn take_message(
+        &mut self,
+        program_id: &ProgramId,
+        payload: &Payload,
+    ) -> Option<Message> {
+        match self.actor_to_mailbox.get_mut(program_id) {
+            None => None,
+            Some(mailbox) => {
+                for i in 0..mailbox.len() {
+                    if mailbox[i].payload.eq(payload) {
+                        return Some(mailbox.remove(i));
+                    }
+                }
+                None
+            }
+        }
```

Comment:
Prefer the func style:

```suggestion
        self.actor_to_mailbox.get_mut(program_id).and_then(|mb| {
            mb.iter()
                .position(|m| m.payload.eq(payload))
                .map(|i| mb.remove(i))
        })
```